backend/types.ts

```diff
@@ -1,16 +1,20 @@
 export type BackendType = "js" | "wasm" | "webgpu";
-export type DataType = "u32" | "i32" | "f32";
+export type DataPrimitive = "u32" | "i32" | "f32";
+export type DataVec = "vec2" | "vec3" | "vec4";
```

Comment:
Instead of this, lets use a string literal type like: `vec${2 | 3 | 4}<${DataPrimitive}>` (Same for matrices: `mat${2 | 3 | 4}x${2 | 3 | 4}<${DataPrimitive}>`)